src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cannot save data", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(sql);
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                Book book = new Book();
+                book.setId(bookId);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            } else {
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to find book with id: " + id);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(sql);
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                long id = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                bookList.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("can`t get book from DataBase ", e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(sql);
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int rowsUpdated = statement.executeUpdate();
+            if (rowsUpdated == 0) {
+                throw new DataProcessingException("Failed to update book with id: " + book.getId());
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update book with id: " + book.getId() + e);
```

Comment:
same here, we should pass it as a separate parameter, because we need the whole original exception info, like stack trace and it's message to be saved, and passed ll the way to the user or dev